src/System.Private.CoreLib/shared/System/Text/Encoding.cs

```diff
@@ -972,8 +956,7 @@ public virtual char[] GetChars(byte[] bytes, int index, int count)
                 charCount = result;
 
             // Copy the data, don't overrun our array!
-            for (index = 0; index < charCount; index++)
-                chars[index] = arrChar[index];
+            new ReadOnlySpan(arrChar).CopyTo(new Span<char>(chars, charCount));
```

Comment:
```suggestion
            new ReadOnlySpan(arrChar, 0, charCount).CopyTo(new Span<char>(chars, charCount));
```

It's possible that `charCount`'s value will be modified by the `GetChars` method (see lines 951 - 956). Make sure to slice the source span appropriately before copying to the destination.